### ISSUE TITLE
go.uuid import is broken

### ISSUE THREAD

**Issue by ineiti** [dedis/onet#660]
After merging of #659 , it turns out that satori/go.uuid is broken and doesn't work as intended with a go-module. As at least one other fork of it, gofrs/uuid, removed go.mod support, I propose go.uuid is replaced with `google/uuid`.

In the meantime, a workaround can be

```
	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b // indirect
```

in onet/go.mod.

**Comment by nkcr** [comment]
No later than yesterday I had to select a library for generating uuids, this article lists some interesting possibilities: https://blog.kowalczyk.info/article/JyRZ/generating-good-unique-ids-in-go.html (we went with rs/xid because we already use rs/zerolog).

**Comment by ineiti** [comment]
And why not `google/uuid`?

**Comment by ineiti** [comment]
And why use uuids at all? Do you really need them? If I'd had to redo that part of the code, I'd use sha256...

**Comment by nkcr** [comment]
UUID can offer interesting properties, such as being more compact, or sortable.

**Comment by ineiti** [comment]
- more compact: it's 16 bytes against 32 bytes
- sortable: both are sortable, no?

Anyway - just wanted to raise the issue that back in the days (when I was young and started with onet) I chose UUID because I thought it will make it easier to interact with other libraries. Up to now it created more problems than it solved...

At least if you're using random UUIDs (version 4). Perhaps a good usage of version 3 or 5 could help. But I did not encounter a use-case where UUIDs would've helped... YMMV.
